### PR TITLE
#159989013 User can reset password

### PR DESCRIPTION
**Pull request by anneKay**
#### What does this PR do?

- User can receive email with a link to reset password
- User can reset password by clicking on the link sent

#### Description of Task to be completed?

    - Add redux functionality
    - Add scss style for password components
    - Add redirection to loginpage after password reset
    - Display error messages for bad requests
    - Write tests

#### How should this be manually tested?
- `git clone https://github.com/andela/fargo-ah-client.git Authors-Haven-Client`
- `npm install`
- `npm run start` the start script
- Go to '/login' path and click on `login` or `register`
- Click on password reset
- Enter valid email 
- Click on link sent through email
- Enter password to reset

#### What are the relevant pivotal tracker stories?

- [#159989014](https://www.pivotaltracker.com/story/show/159989013)

#### Screenshots (if appropriate)

image

image

image

#### Questions:

N/A

### CODE REVIEW

File: src/components/forms/PasswordResetForm.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    });
+    if (Object.keys(errors).length === 0) {
+      this.setState({
+        loading: true,
+      });
+      submit(user.password);
+    }
+  };
+
+  validate = (data) => {
```
[the remark this answers is not in the document]
Comment (ascii-dev) [reply src/components/forms/PasswordResetForm.js]: Fixed this???

File: src/styles/reset.scss [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        width: 60px;
+        height: 40px;
+        position: relative;
+        left: 40%;
+        top: 1em;
+        margin-bottom: 3em;
+      }
+    }
+  }
+}
```
Comment (ascii-dev) [inline src/styles/reset.scss]: New line please?

File: src/styles/resetSuccess.scss [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      }
+      #success-text {
+        margin-top:0em !important;
+      padding: 1em;
+      color: $primary-color;
+      margin-left: 25%;
+      }
+   }
+  }
+}
```
Comment (ascii-dev) [inline src/styles/resetSuccess.scss]: New line 😉

**Review by ascii-dev — CHANGES_REQUESTED** [review]
Kindly fix HoundCI issues and also I think you should add the base endpoint url to env, so that it won't be hard coded every where it's needed just incase of when we deploy. Thanks.

File: src/redux/actions/resetPassword.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import axios from 'axios';
+import actions from './index';
+
+const {
+  RESET_PASSWORD_SUCCESS, RESET_PASSWORD_FAILURE,
+  UPDATE_PASSWORD_FAILURE, UPDATE_PASSWORD_SUCCESS,
+} = actions;
+const BASE_URL_1 = 'https://fargo-ah-staging.herokuapp.com/api';
```
Comment (Oloyedesinmiloluwa) [inline src/redux/actions/resetPassword.js:8]: well done, I think handling this base url in a dynamic way is just appropriate, what about using an environment variable for this?

Comment (anneKay) [reply src/redux/actions/resetPassword.js:8]: Thank you Sinmi for your suggestion, while adding the base URL to env file is a good option that the team is trying to resolve, I suggest we add the base URL to Axios using `axios.defaults.baseURL`

**Review by Oloyedesinmiloluwa — APPROVED** [review]
good

**Review by ascii-dev — APPROVED** [review]

**Review by anuonifade — APPROVED** [review]
LGTM

**Pull request merged**